webapp/graphite/render/hashing.py

```diff
@@ -73,11 +73,8 @@ def __init__(self, nodes, replica_count=100, hash_type='carbon_ch'):
 
   def compute_ring_position(self, key):
     if self.hash_type == 'fnv1a_ch':
-      big_hash = '{:x}'.format(int(fnv32a( str(key) )))
-      if len(big_hash) > 4:
-          small_hash = int(big_hash[:4], 16) ^ int(big_hash[4:], 16)
-      else:
-          small_hash = int(big_hash, 16)
+      big_hash = int(fnv32a(str(key)))
+      small_hash = (big_hash >> 16) ^ (big_hash & 0xffff)
```

Comment:
I see that here we don't have the same python version check as in carbon, we should make them consistent.  The best solution may be to use `big_hash = int(fnv32a(six.binary_type(key)))` for both 2.x and 3.x.

Reply:
Actually that won't work because on 3.x `bytes` required an encoding parameter.  Likely the simplest solution is to use the same arrangement as in carbon.

Reply:
I realized that the carbon approach doesn't properly handle multibyte characters either, if #2168 makes sense to you @iain-buclaw-sociomantic @deniszh then I'll port it over to carbon also.